Bottom: 28253925f4492188054632e5a7067a104adb4106
Top:    9dd2298161eee1ced71c0f0bda82f505ef8566d7
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2015-01-01 17:10:53 +0000

parse_y.y: Try to avoid two shift/reduce and reduce/reduce warnings

TODO: Check this matches up against our expected behavior.

Removing the parsedata section ("I think") has no effect, as elements
hit the alternative | element clause within parsepcb.

HOWEVER... I need to check whether there is any other valid syntax
inside an element file, or any backwards compatibility with old formats
which we may miss with the loss of parsedata.


---

diff --git a/src/parse_y.y b/src/parse_y.y
index ba31cd6..62fcc4b 100644
--- a/src/parse_y.y
+++ b/src/parse_y.y
@@ -131,7 +131,6 @@ static Coord new_units (PLMeasure m);
 
 parse
 		: parsepcb
-		| parsedata
 		| parsefont
 		| error { YYABORT; }
 		;
@@ -240,25 +239,6 @@ parsepcb
 		  }
 		;
 
-parsedata
-		:	{
-					/* reset flags for 'used layers';
-					 * init font and data pointers
-					 */
-				int	i;
-
-				if (!yyData || !yyFont)
-				{
-					Message(_("illegal fileformat\n"));
-					YYABORT;
-				}
-				for (i = 0; i < MAX_LAYER + 2; i++)
-					LayerFlag[i] = false;
-				yyData->LayerN = 0;
-			}
-		 pcbdata
-		;
-
 pcbfont
 		: parsefont
 		|
@@ -706,7 +686,6 @@ pcbdefinition
 		| rats
 		| layer
 		| element
-		| error { YYABORT; }
 		;
 
 via
